ops/testing.py

```diff
@@ -1890,6 +1890,42 @@ def run_action(self, action_name: str,
                 output=action_under_test.output)
         return action_under_test.output
 
+    def set_cloud_spec(self, content: Dict[str, str]):
```

Comment:
Any reason this shouldn't take a proper `CloudSpec` object? They can always use `CloudSpec.from_dict` is they have a dict.